workflows/flows/assemble_study_tasks/make_samplesheets.py

```diff
@@ -107,29 +107,42 @@ def make_samplesheet(
                 }.get(platform, str(platform).lower()),
             ),
             "assembler": SamplesheetColumnSource(
-                lookup_string=[
-                    f"run__metadata__{analyses.models.Run.CommonMetadataKeys.LIBRARY_LAYOUT}",
-                    f"run__metadata__{analyses.models.Run.CommonMetadataKeys.INSTRUMENT_PLATFORM}",
-                ],
                 # PACBIO_SMRT and OXFORD_NANOPORE - flye
                 # SE platform ION_TORRENT         - spades
                 # other SE                        - megahit
                 # all the rest (mostly illumina)  - metaspades
-                renderer=lambda layout, platform: (
+                lookup_string="run__metadata",
+                renderer=lambda metadata: (
                     analyses.models.Assembler.FLYE
-                    if platform
+                    if metadata.get(
+                        analyses.models.Run.CommonMetadataKeys.INSTRUMENT_PLATFORM
+                    )
                     in {
                         analyses.models.Run.InstrumentPlatformKeys.PACBIO_SMRT,
                         analyses.models.Run.InstrumentPlatformKeys.OXFORD_NANOPORE,
                     }
                     else (
                         analyses.models.Assembler.SPADES
-                        if layout == SINGLE_END_LIBRARY_LAYOUT
-                        and platform
+                        if metadata.get(
+                            analyses.models.Run.CommonMetadataKeys.INFERRED_LIBRARY_LAYOUT,
+                            metadata.get(
+                                analyses.models.Run.CommonMetadataKeys.LIBRARY_LAYOUT
+                            ),
+                        )
```

Comment:
Not for this PR... but in case you or I end up adding more such `INFERRED_*` metadata fields, I think we should add an
```python
@property
def inferred_metadata()
```
to the model. Then there we can just overwrite the value of `LIBRARY_LAYOUT` with `INFERRED_LIBRARY_LAYOUT` if it is present. So places like this can just do `run.inferred_metadata.get(LIBRARY_LAYOUT)` and get the value, whether or not it was inferred. All these places should not have to care about where the value came from I think